opnode/rollup/driver/step.go

```diff
@@ -80,6 +78,15 @@ func (d *outputImpl) createNewBlock(ctx context.Context, l2Head eth.L2BlockRef,
 	if err != nil {
 		return l2Head, nil, fmt.Errorf("failed to extend L2 chain: %v", err)
 	}
+	depositStart := 1 // At least the L1 Info deposit
+	for i, tx := range payload.Transactions() {
+		if tx.Type() == types.DepositTxType {
+			depositStart = i + 1 // Want to exclude this deposit
+		} else {
+			// Once done with deposits, assume not more deposit transactions
+			break
```

Comment:
Running through the rest, and actually making sure there are no other deposit txs, and err if there are, would be nice. Then this bug can't happen again. (Also, we should add logging and/or metrics in the derivation function if it ever hits a deposit in a sequencer batch)